src/state.rs

```diff
@@ -1,16 +1,32 @@
 use crate::providers::Providers;
+use crate::relay::RelayClient;
 use crate::store::ClientStore;
 use crate::{env::Config, providers::ProviderKind};
 use build_info::BuildInfo;
 use opentelemetry::metrics::{Counter, UpDownCounter};
 use opentelemetry::sdk::trace::Tracer;
+use std::sync::Arc;
 use tracing_subscriber::prelude::*;
 
+#[derive(Clone)]
 pub struct Metrics {
     pub registered_webhooks: UpDownCounter<i64>,
     pub received_notifications: Counter<u64>,
 }
 
+pub trait State<S>
```

Comment:
Do we really need this trait? I feel like axum should have a simpler way of passing the state down to middleware, but I'm not aware of it...

Reply:
Because we aren't makeing it as a middleware and as an extractor we can only add extra requirements to the `S` state type so I made the trait so we are using the intended was of passing state